pkg/stringid/stringid.go

```diff
@@ -80,16 +86,20 @@ func ValidateID(id string) error {
 
 func init() {
 	// safely set the seed globally so we generate random ids. Tries to use a
-	// crypto seed before falling back to time.
+	// crypto seed (concurrent-safe) before falling back to time (concurrent-unsafe).
 	var seed int64
 	if cryptoseed, err := cryptorand.Int(cryptorand.Reader, big.NewInt(math.MaxInt64)); err != nil {
 		// This should not happen, but worst-case fallback to time-based seed.
-		seed = time.Now().UnixNano()
+		// Attempt to mitigate clashes slightly by including the current process ID.
+		// This is slow and imperfect, but preferable to hanging (indefinately)
+		// for the entropy pool to fill.
+		seed = time.Now().UnixNano() + int64(syscall.Getpid())
```

Comment:
Not a blocker: should we use `rand.Int()` instead of `syscall.Getpid()`.

Reply:
i see. fair point.

Reply:
Also, this code is _initializing_ the `rand.` state. If `rand.Int()` were different enough between processes at this point, we wouldn’t probably need to call `rand.Seed()`. [Note that without this `init()` , `rand.` in all processes is initialized as if with `rand.Seed(1)`, i.e. it is deterministic from the POV of the `rand` state (though it may not be observed deterministically by any single goroutine).